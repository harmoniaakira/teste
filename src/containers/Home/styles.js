import styled from 'styled-components'
import { Typography } from '@material-ui/core'

export const Container = styled.div`
  background: ${({ theme }) => theme.palette.primary.background};
  height: 100vh;
  text-align: center;
`

export const Title = styled(Typography)`
  margin: 0;
  margin-bottom: 10px;
`
