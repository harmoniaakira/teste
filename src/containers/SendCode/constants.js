export const SCREEN_SENDCODE_INTRO = 'SCREEN_SENDCODE_INTRO'
export const SCREEN_SENDCODE_INPUT_TOKEN = 'SCREEN_SENDCODE_INPUT_TOKEN'
export const SCREEN_SENDCODE_INPUT_TOKEN_FAILED = 'SCREEN_SENDCODE_INPUT_TOKEN_FAILED'

export const SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN = 'SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN'
export const SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN = 'SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN'
export const SCREEN_SENDCODE_FEEDBACK_SUCCESS = 'SCREEN_SENDCODE_FEEDBACK_SUCCESS'

export const SCREEN_SENDCODE_ALTER_SEND_METHOD = 'SCREEN_SENDCODE_ALTER_SEND_METHOD'