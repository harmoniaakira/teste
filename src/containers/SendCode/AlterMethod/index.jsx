import { MenuItem } from '@material-ui/core'
import Button from 'components/Button'
import { MobileView } from 'components/Layout/Layout.styles'
import { StyledArrowDown, StyledFormControl, StyledInput, StyledSelect, SytledInputLabel } from 'components/StyledUI/StyledUI.styles'
import Web from 'components/Web'
import { useSendCodeScreen } from 'contexts/SendCodeContext'
import React, { useState } from 'react'
import { maskPhone } from 'utils/maskPhone'
import { SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN, SCREEN_SENDCODE_INPUT_TOKEN_FAILED } from '../constants'
import { ArrowBack, Body, BodyText, BodyTitle, Header, HeaderTitle, WrapperInput } from './AlterMethod.styles'

const SendCodeAlterMethod = () => {

    const { setScreen } = useSendCodeScreen()

    const [phoneNumber, setPhoneNumber] = useState("")

    const handleBackToToken = () => {
        setScreen(SCREEN_SENDCODE_INPUT_TOKEN_FAILED)
    }

    const handlePhoneNumberChange = (event) => {
        const { value } = event.target
        if (value.length > 15) return
        setPhoneNumber(value)
    }

    const handleSendNewCode = () => {
        setScreen(SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN)
    }

    const RenderBody = (
        <Body>
            <BodyTitle>O método informado está correto?</BodyTitle>
            <BodyText>
                Verifique se a informação abaixo está correta ou selecione um novo método de envio.
                O novo código será enviado para ele:
            </BodyText>

            <WrapperInput>
                <StyledInput
                    label="Informe um número de celular válido"
                    fullWidth
                    onChange={handlePhoneNumberChange}
                    value={maskPhone(phoneNumber)}
                />
            </WrapperInput>

            <WrapperInput>
                <StyledFormControl>
                    <SytledInputLabel id="select-sending-method">Escolha por onde enviar o código</SytledInputLabel>
                    <StyledSelect
                        labelId="select-sending-method"
                        IconComponent={StyledArrowDown}
                        style={{ minWidth: 312 }}
                    >
                        <MenuItem value={10}>SMS</MenuItem>
                        <MenuItem value={20} disabled>E-mail</MenuItem>
                        <MenuItem value={30} disabled>WhatsApp</MenuItem>
                    </StyledSelect>
                </StyledFormControl>
            </WrapperInput>

            <Button handleClick={handleSendNewCode}>Reenviar código</Button>
        </Body>
    )


    return (
        <>
            <MobileView>
                <Header onClick={handleBackToToken}>
                    <ArrowBack />
                    <HeaderTitle>
                        Reenviar Código
                    </HeaderTitle>
                </Header>
                {RenderBody}
            </MobileView>
            <Web>
                {RenderBody}
            </Web>
        </>
    )
}

export default SendCodeAlterMethod