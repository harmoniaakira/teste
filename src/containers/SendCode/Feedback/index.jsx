import React from 'react'
import AppHeader from 'components/AppHeader'
import { FlexColumn, MobileView } from 'components/Layout/Layout.styles'
import Web from 'components/Web'
import PropTypes from 'prop-types'
import { SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN, SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN, SCREEN_SENDCODE_FEEDBACK_SUCCESS } from '../constants'
import FeedbackAppraiseToken from './Appraise'
import FeedbackSendToken from './SendToken'
import TokenSuccess from './Success/indes'

const SendCodeFeedBack = ({ feedback }) => {


    const RenderFeedback = () => {
        if (feedback === SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN) {
            return <FeedbackSendToken />
        }

        if (feedback === SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN) {
            return <FeedbackAppraiseToken />
        }

        if (feedback === SCREEN_SENDCODE_FEEDBACK_SUCCESS) {
            return <TokenSuccess />
        }
        return null
    }

    return (
        <>
            <MobileView>
                <AppHeader />
                <RenderFeedback />
            </MobileView>
            <Web>
                <FlexColumn>
                    <RenderFeedback />
                </FlexColumn>
            </Web>
        </>
    )

}

SendCodeFeedBack.propTypes = {
    feedback: PropTypes.oneOf(
        [
            SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN,
            SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN,
            SCREEN_SENDCODE_FEEDBACK_SUCCESS
        ]).isRequired,
}

export default SendCodeFeedBack