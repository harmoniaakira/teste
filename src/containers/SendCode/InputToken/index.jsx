
import Button from 'components/Button'
import { MobileView } from 'components/Layout/Layout.styles'
import Stepper from 'components/Stepper'
import Web from 'components/Web'
import { useSendCodeScreen } from 'contexts/SendCodeContext'
import PropTypes from 'prop-types'
import React, { useState } from 'react'
import { SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN } from '../constants'
import { Form, Input, InputTokenInvalid, InputTokenText, InputTokenTitle, InputWrapper } from './InputToken.styles'

const InputToken = ({ title, text, children, error }) => {
    const { setScreen } = useSendCodeScreen()
    const webInput0 = React.useRef(null);
    const webInput1 = React.useRef(null);
    const webInput2 = React.useRef(null);
    const webInput3 = React.useRef(null);
    const mobileInput0 = React.useRef(null);
    const mobileInput1 = React.useRef(null);
    const mobileInput2 = React.useRef(null);
    const mobileInput3 = React.useRef(null);
    const [inputVal0, setInputVal0] = useState("")
    const [inputVal1, setInputVal1] = useState("")
    const [inputVal2, setInputVal2] = useState("")
    const [inputVal3, setInputVal3] = useState("")

    const handleClick = () => {
        setScreen(SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN)
    }

    const finalInput = () => {
        const str = `${inputVal0}${inputVal1}${inputVal2}${inputVal3}`
        return str
    }

    const focusNextInput = (event, ref, setVal) => {
        if (!event.target.value) setVal(event.target.value)
        if (!/[0-9]/.test(event.target.value)) {
            event.preventDefault();
            return
        }
        setVal(event.target.value)
        if (ref && event.target.value) {
            ref.current.focus();
            ref.current.select();
        }
    }

    const focusPrevInput = (event, ref) => {
        const { key } = event
        if (key === "Backspace" && ref && event.target.value.length === 0) {
            if (ref.current.value.length > 0) {
                event.preventDefault();
            }
            ref.current.select();
        }
    }

    return (
        <>
            <MobileView>
                <Stepper />

                <InputWrapper>
                    <InputTokenTitle>{title}</InputTokenTitle>
                    <InputTokenText>{text}</InputTokenText>
                    <Form $error={error}>
                        <Input
                            ref={mobileInput0}
                            maxLength="1"
                            onKeyDown={focusPrevInput}
                            onChange={(e) => focusNextInput(e, mobileInput1, setInputVal0)}
                            value={inputVal0}
                            key="mobileInput0"
                        />
                        <Input
                            ref={mobileInput1}
                            maxLength="1"
                            onKeyDown={(e) => focusPrevInput(e, mobileInput0)}
                            onChange={(e) => focusNextInput(e, mobileInput2, setInputVal1)}
                            value={inputVal1}
                            key="mobileInput1"
                        />
                        <Input
                            ref={mobileInput2}
                            maxLength="1"
                            onKeyDown={(e) => focusPrevInput(e, mobileInput1)}
                            onChange={(e) => focusNextInput(e, mobileInput3, setInputVal2)}
                            value={inputVal2}
                            key="mobileInput2"
                        />
                        <Input
                            ref={mobileInput3}
                            maxLength="1"
                            onKeyDown={(e) => focusPrevInput(e, mobileInput2)}
                            onChange={(e) => focusNextInput(e, null, setInputVal3)}
                            value={inputVal3}
                            key="mobileInput3"
                        />
                        {error && <InputTokenInvalid>
                            Código inválido
                        </InputTokenInvalid>}
                    </Form>
                    {children}
                    <Button handleClick={handleClick} disabled={finalInput().length < 4}>
                        Confirmar código
                    </Button>
                </InputWrapper>

            </MobileView>

            <Web>
                <InputWrapper>
                    <InputTokenTitle>{title}</InputTokenTitle>
                    <InputTokenText>{text}</InputTokenText>
                    <Form $error={error}>
                        <Input
                            ref={webInput0}
                            maxLength="1"
                            onKeyDown={focusPrevInput}
                            onChange={(e) => focusNextInput(e, webInput1, setInputVal0)}
                            value={inputVal0}
                            key="webInput0"
                        />
                        <Input
                            ref={webInput1}
                            maxLength="1"
                            onKeyDown={(e) => focusPrevInput(e, webInput0)}
                            onChange={(e) => focusNextInput(e, webInput2, setInputVal1)}
                            value={inputVal1}
                            key="webInput1"
                        />
                        <Input
                            ref={webInput2}
                            maxLength="1"
                            onKeyDown={(e) => focusPrevInput(e, webInput1)}
                            onChange={(e) => focusNextInput(e, webInput3, setInputVal2)}
                            value={inputVal2}
                            key="webInput2"
                        />
                        <Input
                            ref={webInput3}
                            maxLength="1"
                            onKeyDown={(e) => focusPrevInput(e, webInput2)}
                            onChange={(e) => focusNextInput(e, null, setInputVal3)}
                            value={inputVal3}
                            key="webInput3"
                        />
                        {error && <InputTokenInvalid>
                            Código inválido
                        </InputTokenInvalid>}
                    </Form>
                    {children}
                    <Button handleClick={handleClick} disabled={finalInput().length < 4}>
                        Confirmar código
                    </Button>
                </InputWrapper>
            </Web>
        </>
    )
}

InputToken.propTypes = {
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    error: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ])
}

InputToken.defaultProps = {
    children: null,
    error: false
}

export default InputToken
