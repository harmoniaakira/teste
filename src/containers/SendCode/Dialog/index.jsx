import { Dialog, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
import { DialogActionsBetween, DialogBtn } from 'components/StyledUI/StyledUI.styles';
import { useSendCodeDialog, useSendCodeScreen } from 'contexts/SendCodeContext';
import React, { useState } from 'react';
import { SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN } from '../constants';
import { Flex } from '../SendCode.styles';

const SendCodeDialog = () => {
    const [phase, setPhase] = useState(0)
    const { isDialogSendCodeOpen, setIsDialogSendCodeOpen } = useSendCodeDialog()
    const { setScreen } = useSendCodeScreen()

    const handleClose = () => {
        setPhase(0)
        setIsDialogSendCodeOpen(false)
    }

    const Content = () => {
        switch (phase) {
            default:
                return "Certifique-se que seu cliente está em posse do celular informado. Deseja seguir com o código de validação?"

            case 1:
                return "Sem a validação do código de segurança, a solicitação estará sujeita a análise diferenciada da proposta. Deseja seguir sem o código de segurança?"

            case 2:
                return "Você poderá continuar a partir dessa etapa quando o cliente estiver com o celular em mãos. ;)"
        }
    }

    const handleAppraiseToken = () => {
        setPhase(0)
        setIsDialogSendCodeOpen(false)
        setScreen(SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN)
    }

    const RenderBtns = () => {
        switch (phase) {
            default:
                return <>
                    <DialogBtn onClick={() => setPhase(1)}>
                        Voltar
                    </DialogBtn>
                    <DialogBtn onClick={handleAppraiseToken}>
                        Continuar
                    </DialogBtn>
                </>
            case 1:
                return <>
                    <DialogBtn onClick={() => setPhase(2)}>
                        Voltar
                    </DialogBtn>
                    <DialogBtn onClick={handleClose} disabled>
                        Continuar
                    </DialogBtn>
                </>
            case 2:
                return <>
                    <Flex />
                    <DialogBtn onClick={handleClose}>
                        Fechar
                    </DialogBtn>
                </>
        }
    }

    return (
        <Dialog
            open={isDialogSendCodeOpen}
            onClose={handleClose}
        >
            <DialogTitle>Atenção</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    <Content />
                </DialogContentText>
            </DialogContent>
            <DialogActionsBetween>
                <RenderBtns />
            </DialogActionsBetween>
        </Dialog>
    )
}

export default SendCodeDialog