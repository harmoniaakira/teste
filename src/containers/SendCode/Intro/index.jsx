import { MenuItem } from '@material-ui/core'
import avatar from 'assets/images/SendCodeView/avatar.png'
import step1 from 'assets/images/SendCodeView/step1.svg'
import step2 from 'assets/images/SendCodeView/step2.svg'
import step3 from 'assets/images/SendCodeView/step3.svg'
import AppHeader from 'components/AppHeader'
import Steps from 'components/Steps'
import Button from 'components/Button'
import { MobileView } from 'components/Layout/Layout.styles'
import { StyledArrowDown, StyledFormControl, StyledSelect, SytledInputLabel } from 'components/StyledUI/StyledUI.styles'
import Web from 'components/Web'
import { useSendCodeDialog } from 'contexts/SendCodeContext'
import React from 'react'
import SendCodeDialog from '../Dialog'
import { Avatar, Content, HeaderText, HeaderTitle, WrapperSendCode } from '../SendCode.styles'

const steps = [
    {
        id: 1,
        stepIcon: step1,
        stepText: "Selecione o método de envio"
    },
    {
        id: 2,
        stepIcon: step2,
        stepText: "Envie e verifique se o cliente recebeu o código no método solicitado"
    },
    {
        id: 3,
        stepIcon: step3,
        stepText: "Digite o código corretamente"
    }
]

const SendCodeIntro = () => {

    const { setIsDialogSendCodeOpen } = useSendCodeDialog()

    const handleClick = () => {
        setIsDialogSendCodeOpen(true)
    }

    return (
        <>
            <MobileView>
                <AppHeader avatar={avatar} />
                <HeaderTitle>Validação do código</HeaderTitle>
                <HeaderText>
                    Enviaremos um código de segurança para o cliente. Preencha seguindo as
                    intruções:
                </HeaderText>
                <Steps steps={steps} />

                <StyledFormControl style={{ margin: '0 auto' }}>
                    <SytledInputLabel id="select-sending-method">Escolha por onde enviar o código</SytledInputLabel>
                    <StyledSelect
                        labelId="select-sending-method"
                        IconComponent={StyledArrowDown}
                        style={{ minWidth: 312, marginBottom: 60 }}
                    >
                        <MenuItem value={10}>SMS</MenuItem>
                        <MenuItem value={20}>E-mail</MenuItem>
                        <MenuItem value={30}>WhatsApp</MenuItem>
                    </StyledSelect>
                </StyledFormControl>
                <div style={{ paddingBottom: 60 }}>
                    <Button handleClick={handleClick}>Enviar código</Button>
                </div>
            </MobileView>

            <Web>
                <Content>
                    <WrapperSendCode>
                        <Avatar src={avatar} />
                        <HeaderTitle>Validação do código</HeaderTitle>
                        <HeaderText>
                            Enviaremos um código de segurança para o cliente. Preencha seguindo as
                            intruções:
                        </HeaderText>
                        
                        <Steps steps={steps} />

                        <StyledFormControl>
                            <SytledInputLabel id="select-sending-method">Possui necessidade especial? </SytledInputLabel>
                            <StyledSelect
                                labelId="select-sending-method"
                                IconComponent={StyledArrowDown}
                                style={{ minWidth: 312, marginBottom: 60 }}
                            >
                                <MenuItem value={10}>SMS</MenuItem>
                                <MenuItem value={20}>E-mail</MenuItem>
                                <MenuItem value={30}>WhatsApp</MenuItem>
                            </StyledSelect>
                        </StyledFormControl>
                    </WrapperSendCode>

                    <Button handleClick={handleClick}>Enviar código</Button>
                </Content>
            </Web>
            <SendCodeDialog />
        </>
    )
}

export default SendCodeIntro