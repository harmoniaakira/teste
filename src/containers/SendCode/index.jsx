import Button from 'components/Button'
import { ContainerPage } from 'components/Layout/Layout.styles'
import SwitchFeedback from 'components/SwitchFeedback'
import { useSendCodeScreen } from 'contexts/SendCodeContext'
import React from 'react'
import {
  SCREEN_SENDCODE_ALTER_SEND_METHOD, SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN, SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN,
  SCREEN_SENDCODE_FEEDBACK_SUCCESS, SCREEN_SENDCODE_INPUT_TOKEN, SCREEN_SENDCODE_INPUT_TOKEN_FAILED, SCREEN_SENDCODE_INTRO
} from './constants'
import FeedbackSendToken from './Feedback/SendToken'
import TokenSuccess from './Feedback/Success/indes'
import InputTokenDefault from './InputToken/Default'
import InputTokenUnrecognized from './InputToken/Unrecognized'
import SendCodeIntro from './Intro'
import { Apagar } from './SendCode.styles'


const SendCode = () => {

  const { screen, setScreen } = useSendCodeScreen()

  // TODO: ARRUMAR
  const feeds = [
    {
      id: SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN,
      component: (
        <>
          <FeedbackSendToken />
        </>
      )
    },
    {
      id: SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN,
      component: (
        <>
          <FeedbackSendToken />
          <Apagar>
            <Button handleClick={() => setScreen(SCREEN_SENDCODE_FEEDBACK_SUCCESS)}> Sucesso </Button>
            <Button handleClick={() => setScreen(SCREEN_SENDCODE_INPUT_TOKEN_FAILED)}> Erro </Button>
          </Apagar>
        </>
      )
    },
    {
      id: SCREEN_SENDCODE_FEEDBACK_SUCCESS,
      component: <TokenSuccess />
    }
  ]

  const switchContent = () => {

    switch (screen) {
      default:
      case SCREEN_SENDCODE_INTRO:
        return <SendCodeIntro />

      case SCREEN_SENDCODE_FEEDBACK_SENDING_TOKEN:
      case SCREEN_SENDCODE_FEEDBACK_ANALYZING_TOKEN:
      case SCREEN_SENDCODE_FEEDBACK_SUCCESS:
        return <SwitchFeedback feedbackKey={screen} feeds={feeds} />


      case SCREEN_SENDCODE_INPUT_TOKEN:
        return <InputTokenDefault />


      case SCREEN_SENDCODE_INPUT_TOKEN_FAILED:
      case SCREEN_SENDCODE_ALTER_SEND_METHOD:
        return <InputTokenUnrecognized />
    }
  }

  return (
    <ContainerPage>
      {switchContent()}
    </ContainerPage>
  )
}

export default SendCode;
