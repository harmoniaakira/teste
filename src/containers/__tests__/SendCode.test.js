import Button from 'components/Button';
import SendCodeDialog from 'containers/SendCode/Dialog';
import SendCodeFeedBack from 'containers/SendCode/Feedback';
import FeedbackAppraiseToken from 'containers/SendCode/Feedback/Appraise';
import FeedbackSendToken from 'containers/SendCode/Feedback/SendToken';
import TokenSuccess from 'containers/SendCode/Feedback/Success/indes';
import SendCode from 'containers/SendCode/index.jsx';
import InputTokenDefault from 'containers/SendCode/InputToken/Default';
import InputTokenUnrecognized from 'containers/SendCode/InputToken/Unrecognized';
import SendCodeIntro from 'containers/SendCode/Intro';
import { DialogBtn } from 'containers/SendCode/SendCode.styles';
import { render, shallow } from 'enzyme';
import React from 'react';

describe('SendCode', () => {
  const sendCode = shallow(<SendCode />);

  it('should match snapshot', () => {
    expect(sendCode).toMatchSnapshot()
  })
})


describe('SendCode/InputTokenDefault', () => {

  const inputTokenDefault = shallow(<InputTokenDefault />);

  it("should render", () => {
    expect(inputTokenDefault).toHaveLength(1);
  });

})

describe('SendCode/InputTokenUnrecognized', () => {

  const inputTokenUnrecognized = shallow(<InputTokenUnrecognized />);

  it("should render", () => {
    render(<InputTokenUnrecognized />)
    expect(inputTokenUnrecognized).toHaveLength(1);
    expect(inputTokenUnrecognized.find('RenderReSend')).toHaveLength(1);
  });

})

describe('SendCode/SendCodeDialog', () => {

  const sendCodeDialog = shallow(<SendCodeDialog />);

  it("should render", () => {
    expect(sendCodeDialog.find('RenderBtns')).toHaveLength(1);
    expect(sendCodeDialog).toHaveLength(1);
  });

})

describe('SendCode/SendCodeFeedBack', () => {

  render(<SendCodeFeedBack feedback={"SENDCODE_FEEDBACK_SENDING_TOKEN"} />);
  render(<SendCodeFeedBack feedback={"SENDCODE_FEEDBACK_ANALYZING_TOKEN"} />);
  render(<SendCodeFeedBack feedback={"SENDCODE_FEEDBACK_SUCCESS"} />);

})


describe('SendCode/SendCodeIntro', () => {

  const sendCodeIntro = shallow(<SendCodeIntro />);

  it("should render", () => {
    // expect(sendCodeIntro.find('RenderBtns')).toHaveLength(1);
    sendCodeIntro.find(Button).at(0).props().handleClick()
    expect(sendCodeIntro).toHaveLength(1);
  });

})