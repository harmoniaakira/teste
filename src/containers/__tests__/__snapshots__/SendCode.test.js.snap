// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendCode should match snapshot 1`] = `
<styled.div>
  <RenderContent />
</styled.div>
`;
